### PR TITLE
MISUV-543: create controller, connector, model and parser for charge…

### PR DESCRIPTION
**Pull request by scullythornton1**
…s and payments

### CODE REVIEW

File: app/models/financialDetails/Charge.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+object Charge {
+
+  implicit val reads: Reads[Charge] = (
+    (JsPath \ "taxYear").readNullable[String] and
+      (JsPath \ "documentId").readNullable[String] and
+      (JsPath \ "documentDate").readNullable[String] and
+      (JsPath \ "documentDescription").readNullable[String] and
+      (JsPath \ "totalAmount").readNullable[BigDecimal] and
+      (JsPath \ "documentOutstandingAmount").readNullable[BigDecimal] and
+      (JsPath \ "subItem").readNullable[Seq[SubItem]]
```
Comment (JSimmons072019) [inline app/models/financialDetails/Charge.scala:40]: (JsPath \ **"items"**).readNullable[Seq[SubItem]]

File: app/models/financialDetails/Payment.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+object Payment {
+
+  implicit val reads: Reads[Payment] = for {
+    paymentLot <- (JsPath \ "paymentLot").readNullable[String]
+    paymentLotItem <- (JsPath \ "paymentLotItem").readNullable[String]
+    amount <- (JsPath \ "paymentAmount").readNullable[BigDecimal]
+    method <- (JsPath \ "paymentMethod").readNullable[String]
+    transactionDate <- (JsPath \ "valueDate").readNullable[String]
+    subItem <- (JsPath \ "subItem").readNullable[Seq[SubItem]]
```
Comment (JSimmons072019) [inline app/models/financialDetails/Payment.scala:35]: **"items"**

File: app/models/financialDetails/SubItem.scala [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                   paymentId: Option[String])
+
+object SubItem {
+
+  val empty: SubItem = SubItem(None, None, None, None, None, None, None, None, None)
+
+  implicit val writes: OWrites[SubItem] = Json.writes[SubItem]
+
+  implicit val reads: Reads[SubItem] = for {
+    subItemId <- (JsPath \ "subItem").readNullable[String](Reads.of[String].filter(subItemJsonError)(isIntString))
```
Comment (JSimmons072019) [inline app/models/financialDetails/SubItem.scala:39]: items

File: it/controllers/FinancialDetailChargesControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  val chargeJson: JsObject = Json.obj(
+    "financialDetails" -> Json.arr(
+      Json.obj(
+        "taxYear" -> "2018",
+        "documentId" -> "transactionId",
+        "documentDate" -> "transactionDate",
+        "documentDescription" -> "type",
+        "totalAmount" -> 1000.00,
+        "documentOutstandingAmount" -> 500.00,
+        "subItem" -> Json.arr(
```
Comment (JSimmons072019) [inline it/controllers/FinancialDetailChargesControllerISpec.scala:84]: items

File: it/controllers/FinancialDetailChargesControllerISpec.scala [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      Json.obj(
+        "taxYear" -> "2019",
+        "documentId" -> "transactionId2",
+        "documentDate" -> "transactionDate2",
+        "documentDescription" -> "type2",
+        "totalAmount" -> 2000.00,
+        "documentOutstandingAmount" -> 200.00,
+        "subItem" -> Json.arr(
+          Json.obj(
+            "subItem" -> "2",
```
Comment (JSimmons072019) [inline it/controllers/FinancialDetailChargesControllerISpec.scala:108]: "items"

File: it/controllers/FinancialDetailPaymentsControllerISpec.scala [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    "financialDetails" -> Json.arr(
+      Json.obj(
+        "paymentLot" -> "paymentLot",
+        "paymentLotItem" -> "paymentLotItem",
+        "paymentAmount" -> 1000.00,
+        "paymentMethod" -> "paymentMethod",
+        "valueDate" -> "valueDate",
+        "subItem" -> Json.arr(
+          Json.obj(
+            "subItem" -> "1",
```
Comment (JSimmons072019) [inline it/controllers/FinancialDetailPaymentsControllerISpec.scala:80]: items

File: it/controllers/FinancialDetailPaymentsControllerISpec.scala [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+      ),
+      Json.obj(
+        "paymentLot" -> "paymentLot2",
+        "paymentLotItem" -> "paymentLotItem2",
+        "paymentAmount" -> 2000.00,
+        "paymentMethod" -> "paymentMethod2",
+        "valueDate" -> "valueDate2",
+        "subItem" -> Json.arr(
+          Json.obj(
+            "subItem" -> "2",
```
Comment (JSimmons072019) [inline it/controllers/FinancialDetailPaymentsControllerISpec.scala:101]: items

File: app/models/financialDetails/Charge.scala (hunk 1 above)
Comment (scullythornton1) [reply app/models/financialDetails/Charge.scala]: done

File: app/models/financialDetails/Payment.scala (hunk 2 above)
Comment (scullythornton1) [reply app/models/financialDetails/Payment.scala]: done

**Pull request merged**